test/e2e/blockaid/toggle-settings.spec.js

```diff
@@ -0,0 +1,75 @@
+const { strict: assert } = require('assert');
+const {
+  withFixtures,
+  unlockWallet,
+  openDapp,
+  defaultGanacheOptions,
+  getWindowHandles,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+
+describe('Blockaid Settings', function () {
+  it('should not show blockaid UI when toggle is off', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await unlockWallet(driver);
+
+        await driver.clickElement(
+          '[data-testid="account-options-menu-button"]',
+        );
+
+        await driver.clickElement({ text: 'Settings', tag: 'div' });
+        await driver.clickElement({ text: 'Experimental', tag: 'div' });
+
+        await openDapp(driver);
+        await driver.clickElement('#signPermit');
+      },
+    );
+  });
+
+  it('should show the blockaid UI when the toggle is on', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await unlockWallet(driver);
+
+        await driver.clickElement(
+          '[data-testid="account-options-menu-button"]',
+        );
+
+        await driver.clickElement({ text: 'Settings', tag: 'div' });
+        await driver.clickElement({ text: 'Experimental', tag: 'div' });
+
+        await driver.clickElement(
+          '[data-testid="transaction-security-check-blockaid"] .toggle-button > div',
+        );
+
+        await openDapp(driver);
+        await driver.clickElement('#signPermit');
+        const windowHandles = await getWindowHandles(driver, 3);
+        await driver.switchToWindow(windowHandles.popup);
+
+        const blockaidResponseTitle = '[data-testid="mm-banner-base-title"]';
+        const exists = await driver.isElementPresent(blockaidResponseTitle);
```

Comment:
I'm wondering how this test is passing if we don't have the PPOM envars setup in the flask build :thinking:

Reply:
This test checks for blockaid fallback response until the envars are confirmed to be working. So yeah, we can consider that a blocker for this PR.

Reply:
@blackdevelopa thank you for the explanation